aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,230 @@
+
+You must have a view with the proper JavaScript script references in place for client side validation to work. 
+
+.. code-block:: html
+
+  <script src="~/lib/jquery/jquery.js"></script>
+  <script src="~/lib/jquery-validation/dist/jquery.validate.js"></script>
+  <script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js"></script>
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages using JavaScript. When you use MVC to render form elements from a model using `Tag Helpers <https://docs.asp.net/en/latest/mvc/views/tag-helpers/index.html>`_ or `HTML helpers <https://docs.asp.net/en/latest/mvc/views/html-helpers.html>`_ it will add HTML 5 `data- attributes <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element. You can display validation errors on the client using the relevant tag helpers as shown here:
+
+.. code-block:: html
+ :emphasize-lines: 7, 8
+
+  <form asp-action="Create">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+        <div class="form-group">
+          <label asp-for="ReleaseDate" class="col-md-2 control-label"></label>
+          <div class="col-md-10">
+            <input asp-for="ReleaseDate" class="form-control" />
+            <span asp-validation-for="ReleaseDate" class="text-danger" />
+          </div>		
+      </div>	
+    </div>  
+  </form>
+
+The tag helpers above render the HTML below. Notice that the ``data-`` attributes in the HTML output correspond to the data annotations for the ``ReleaseDate`` property. 
+
+.. code-block:: html
+
+  <form action="/movies/Create" method="post">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <div class="text-danger"></div>	
+      <div class="form-group">
+        <label class="col-md-2 control-label" for="ReleaseDate">ReleaseDate</label>
+        <div class="col-md-10">
+          <input class="form-control" type="datetime" 
+          data-val="true" data-val-required="The ReleaseDate field is required." 
+          id="ReleaseDate" name="ReleaseDate" value="" />		
+          <span class="text-danger field-validation-valid" 
+          data-valmsg-for="ReleaseDate" data-valmsg-replace="true" />
+        </div>
+      </div>	  
+      </div>
+  </form>
+            
+Client-side validation prevents submission until the form is valid. The Submit button runs JavaScript that either submits the form or displays error messages. 
```

Comment:
@rachelappel the paragraph you added below covers the bullets I listed above, thx. @danroth27 do we need more here or perhaps more and better words than my suggestions?